layer/store/geo.ts

```diff
@@ -0,0 +1,207 @@
+        throw new HttpRequestException(new Error((e as any).message), {
+          contextModule: 'region'
+        })
+      }
+    },
+
+    async fetchVPNStatus() {
+      const sharedGeoStore = useSharedGeoStore()
+
+      if (!sharedGeoStore.ipAddress) {
+        await sharedGeoStore.fetchIpAddress()
+      }
+
+      const httpClient = new HttpClient('https://vpnapi.io/', { timeout: 1000 })
+
+      try {
+        const response = (await httpClient.get(
+          `api/${sharedGeoStore.ipAddress}`,
+          {
+            key: PROXY_DETECTION_API_KEY
+          }
+        )) as {
+          data: {
+            security: {
+              vpn: boolean
+              proxy: boolean
+              tor: boolean
+              relay: boolean
+            }
+            location: {
+              country_code: string
+            }
+          }
+        }
+
+        if (!response.data) {
+          sharedGeoStore.$patch({
+            vpnDetected: false
+          })
+
+          return
+        }
+
+        const { security } = response.data
+
+        const vpnDetected =
+          security.proxy || security.vpn || security.tor || security.relay
+
+        sharedGeoStore.$patch({
+          vpnDetected
+        })
+      } catch (e: unknown) {
+        sharedGeoStore.$patch({
+          vpnDetected: false
+        })
+      }
+    },
+
+    async fetchUserCountryFromBrowser() {
+      const sharedGeoStore = useSharedGeoStore()
+
+      const position = (await new Promise((resolve, reject) =>
+        navigator.geolocation.getCurrentPosition(resolve, reject)
+      ).catch(() => {
+        return {
+          longitude: '',
+          latitude: ''
+        }
+      })) as {
+        coords: {
+          longitude: string
+          latitude: string
+        }
+      }
+
+      if (position.coords.longitude === '' || position.coords.latitude === '') {
+        return
+      }
+
+      const googleMapsHttpClient = new HttpClient(
+        'https://maps.googleapis.com/maps/api/geocode/'
+      )
+      const GOOGLE_MAPS_SUFFIX = `json?latlng=${position.coords.longitude},${position.coords.latitude}&sensor=false&key=${GOOGLE_MAPS_KEY}`
+
+      try {
+        const response = (await googleMapsHttpClient.get(
+          GOOGLE_MAPS_SUFFIX
+        )) as {
+          data: {
+            results: {
+              address_components: { types: string[]; short_name: string }[]
+            }[]
+          }
+        }
+
+        const [results] = response.data.results
+
+        const country = results.address_components.find((component) =>
+          component.types.includes('country')
+        )
+
+        sharedGeoStore.$patch({
+          browserCountry: country?.short_name || ''
+        })
+      } catch (e: unknown) {
+        // silently throw
+      }
+    },
+
+    async fetchUserLocation() {
+      const sharedGeoStore = useSharedGeoStore()
+
+      await sharedGeoStore.fetchGeoLocation()
+
+      if (VPN_CHECKS_ENABLED) {
```

Comment:
I think this would make more sense as an early return, i.e. 
``` 
if (!VPN_CHECKS_ENABLED) {
   return
}
```

Reply:
also, why do we need this `VPN_CHECKS_ENABLED` flag? instead of using `GEO_IP_RESTRICTIONS_ENABLED` flag?